### PR TITLE
Basic Jax implementation

### PR DESCRIPTION
**Pull request by jmarshrossney**
This PR adds a very basic implementation of the Lotka-Volterra system in Jax.

The idea is just to start playing around with fitting the solutions of ODE systems to data by adjusting the parameters of the ODE (here, the 4 coefficients and potentially the initial conditions).

It's mostly for me to gain familiarity with Jax, diffrax and  equinox, and perhaps some differences between training models based on neural networks, and this kind of parameter optimisation.

### CODE REVIEW

**Comment by jmarshrossney** [comment]
Just to test the tools (autodiff etc) I've added a script that minimises the mean sq error between the model and data time series' using an explicit first-order gradient descent loop with a fixed learning rate.

I expected that this naive loss would not work well, since it would be very sensitive to initial conditions, and probably fall into local optima with the wrong oscillation period.

I held the initial conditions fixed and allowed the four coefficients to be optimised.

It doesn't converge when the parameters are initialised far from their 'true' values (probably SGD would do much better here, but it makes sense to look for a better loss function first). 

However, just as a sanity check, it does converge when the parameters are initialised close to the correct values - the loss surface is nice and convex here.

fit

loss
